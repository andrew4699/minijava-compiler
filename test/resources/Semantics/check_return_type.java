class A {

  public static void main(String[] args) {
    System.out.println(new B().f());
  }
}

class B {
  int k;
  B b;

  public int f() {
    boolean x;
    x = this.g();
    return 1;
  }

  public boolean g() {
    return 0;
  }

  public B h() {
    return new D();
  }

  public D i() {
    return new B();
  }
}

class C extends B {

}

class D extends C {

}