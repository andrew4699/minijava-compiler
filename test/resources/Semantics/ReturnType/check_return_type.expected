[ReturnTypeVisitor - 18]: B.g returned int but expects boolean
[ReturnTypeVisitor - 26]: B.i returned B but expects D
